src/frontend/js/widgets/Dashboard/components/DashboardItem/Order/DashboardItemOrder.tsx

```diff
@@ -79,34 +152,43 @@ export const DashboardItemOrder = ({
   const intl = useIntl();
   const query = useCourseProduct(course.code, { productId: order.product });
   const product = query?.item?.product;
+
   const getRoutePath = getDashboardRoutePath(useIntl());
 
   return (
-    <DashboardItem
-      data-testid={`dashboard-item-order-${order.id}`}
-      title={product?.title ?? ''}
-      code={'Ref. ' + course.code}
-      imageUrl={course.cover?.src}
-      footer={
-        <div className="dashboard-item-order__footer">
-          <div className="dashboard-item__block__status">
-            <Icon name={IconTypeEnum.SCHOOL} />
-            <OrderStateMessage order={order} />
-          </div>
-          {showDetailsButton && (
-            <RouterButton
-              className="dashboard-item__button"
-              color="transparent-darkest"
-              href={getRoutePath(LearnerDashboardPaths.ORDER, { orderId: order.id })}
-              data-testid="dashboard-item-order__button"
-            >
-              {intl.formatMessage(messages.accessCourse)}
-            </RouterButton>
-          )}
-        </div>
-      }
-    >
-      <>
+    <div className="dashboard-item-order">
+      {writable && !order.contract?.signed_on && (
+        <DashboardItemOrderContract order={order} product={product} />
+      )}
+      <DashboardItem
+        data-testid={`dashboard-item-order-${order.id}`}
+        title={product?.title ?? ''}
+        code={'Ref. ' + course.code}
+        imageUrl={course.cover?.src}
+        footer={
+          <>
+            <div className="dashboard-item-order__footer">
+              <div className="dashboard-item__block__status">
+                <Icon name={IconTypeEnum.SCHOOL} />
+                <OrderStateMessage order={order} />
+              </div>
+              {showDetailsButton && (
+                <RouterButton
+                  className="dashboard-item__button"
+                  color="transparent-darkest"
+                  href={getRoutePath(LearnerDashboardPaths.ORDER, { orderId: order.id })}
+                  data-testid="dashboard-item-order__button"
+                >
+                  {intl.formatMessage(messages.accessCourse)}
+                </RouterButton>
+              )}
+            </div>
+            {!writable && !order.contract?.signed_on && (
```

Comment:
This condition could cause a bug. All order without a contract are showing the banner "contract to sign"

<img width="1936" alt="Capture d’écran 2023-10-24 à 17 01 10" src="https://github.com/openfun/richie/assets/9628870/c17e2e15-4714-4972-8cc6-af415c6b2b32">